Sources/BitcoinKit/Wallet/Protocol/AddressProvider.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  AddressProvider.swift
+//
+//  Copyright © 2018 BitcoinKit developers
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public protocol AddressProvider {
+    // GET API: reload utxos
+    func reload(keys: [PrivateKey], completion: (([Address]) -> Void)?)
```

Comment:
I think `convert`  is better as this function name.
Besides, I doubt that this protocol should be under `Wallet` directory.
I think just extension of `PrivateKey` would be better, but still not best idea.

Reply:
No, converting `PrivateKey` to `Address` is not enough. 
Mainly this protocol is for P2SH.

Someone may want to compute new P2SH addresses using the keys outside as well as the keys inside by using API or some other way.